### PR TITLE
Kudzu is now more intelligent about where it tries to grow

### PR DESCRIPTION
**Pull request by Crazylemon64**
This makes it spread monstrously fast; none shall be spared.
This might be better as a specific mutation, instead of standard, extremely deadly behavior - this removes one of the primary inhibitors on kudzu growth.

:cl:Crazylemon
experiment: Space vines now have a mutation that lets them grow in a more intelligent manner
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
I do kinda fear the impacts of this, given how ridiculous Kudzu can be in the right scenario---though, as you said, as a specific mutation? Yeah, that'd be pretty cool, I suppose.

File: code/modules/events/spacevine.dm [hunk 1]
```diff
@@ -532,15 +535,22 @@
...
-				master.spawn_spacevine_piece(stepturf, src)
+	var/list/dir_list = cardinal.Copy()
+	while(dir_list.len)
+		var/direction = pick(dir_list)
+		dir_list -= direction
+		var/turf/stepturf = get_step(src,direction)
+		var/spread_success = FALSE
+		for(var/datum/spacevine_mutation/SM in mutations)
+			spread_success |= SM.on_spread(src, stepturf) // If this returns 1, spreading succeeded
+			stepturf = get_step(src,direction) //in case turf changes, to make sure no runtimes happen
```
Comment (Krausus) [inline code/modules/events/spacevine.dm]: No need for this; turfs are only ever actually deleted if the world size is reduced.

Comment (Crazylemon64) [inline code/modules/events/spacevine.dm]: Yeah, the ref should point to the same datum - this is just a holdover from re-indenting existing code

**Comment by Crazylemon64** [comment]
It's a mutation now, not default behavior.

The implementation is super janky though, oops

**Comment by Krausus** [comment]
> This makes it spread monstrously fast

Uhh, yeah, roughly 4 times as fast. Are you sure this won't be totally broken? How fast does Kudzu normally grow?

It would probably make more sense for this to be the default behavior, with a probability attached to whether it spreads or not.

**Comment by Anticept** [comment]
Kudzu already grows fast as hell. 4x faster? that will be unbelieveably hard to stop.

**Comment by Crazylemon64** [comment]
This is mitigated by this being a mutation - meaning, other vines will exist, and the faster growth rates will be spent on those non-virulently spreading vines, too. That way, while the virulent vines will be more able to spread through narrow corridors, they won't be able to take advantage of their true power to its full extent, unless you did something dumb like prune everything except the virulently-spreading ones.

**Comment by Fox-McCloud** [comment]
What could possible go wrong.

**Pull request merged**